### PR TITLE
Migrate Face, Barcode and Text Recognition to Firebase MLKit (iOS - text, Android - all)

### PR DESCRIPTION
**Pull request by dov11**

### CODE REVIEW

**Comment by dov11** [comment]
> does this affect face detection?

This PR only touches text recognition, face detection is not affected

> does this requires that everybody install firebase library?

Installation of react-native-firebase is not necessary, but in order to use MLKit one [needs](https://firebase.google.com/docs/ml-kit/ios/recognize-text#before-you-begin) to setup a firebase project and add firebase sdk (either via cocoapods or by [manually adding a library](https://firebase.google.com/docs/ios/setup#frameworks). 
PR includes amendments to README explaining the required steps.

**Comment by sibelius** [comment]
what about android counter part?

**Comment by dov11** [comment]
I have added migration of Text recognition to MLKit for android, will need to migrate barcode and facedetector so that GMV can be removed completely from this new installation option. Will try to do it soon.

**Comment by dov11** [comment]
I have migrated Android Face and Barcode Detectors to MLkit as well. For now only feature parity with GMV is achieved, as I have not implemented such new features as contours for Face detector yet (Should not be too difficult, but I feel that this implementation is somewhat outside of the scope of this PR).
Also I believe this PR fixes incorrect bounds of text, face and barcode boxes (e.g.  #2057). Tested Face and Text recognition on Galaxy S7 and boxes' positions look better to me.

**Comment by sibelius** [comment]
are you going to send PR to android face and barcode detection using MLKit?

or they are almost tidy together?

**Comment by dov11** [comment]
My most recent push to this PR includes migration of face and barcode detection using MLKit
So, this PR includes:

Android: Face, Barcode and Text Detection (user can use either MLkit or deprecated GMV if they don't want to setup Firebase project)

iOS: Text Detection is migrated to MLKit (optional installation as before)

**Comment by sibelius** [comment]
does anything changes on javascript side? does we need to update types?

**Comment by dov11** [comment]
Javascript side should stay the same, changes that I made should not affect types either

**Comment by sibelius** [comment]
Is example updated with this changes?

**Comment by sibelius** [comment]
Do u plan o work on face recognition and barcode for iOS?

**Comment by dov11** [comment]
Example app is updated to use general flavor(GMV) for now. Readme updated with instructions.
 I tested MLKit implementation in my own clean repo for which I created a firebase project. Example app of react-native-camera might be set up as well, but I did not give it a try.

I might give face and barcode detector for iOS a shot as well, but most likely will bring them as optional Cocoapods installation, so that these changes will be independent from this PR, and it does not need to wait for the iOS implementation.

File: README.md [hunk 1]
```diff
@@ -256,7 +256,31 @@ pod 'react-native-camera', path: '../node_modules/react-native-camera', subspecs
   'TextDetector'
 ]
 ```
-*Note:* Text recognition is available only via CocoaPods Path. If you have issues with duplicate symbols you will need to enable dead code stripping option in your Xcode (Target > Build Settings > search for "Dead code stripping") see: https://github.com/firebase/quickstart-ios/issues/487#issuecomment-415313053.
+
+###### Additional steps for Text Recognition
+Text recognition for iOS uses Firebase MLKit which requires seting up Firebase project for your app.
```
Comment (n1ru4l) [inline README.md]:
```suggestion
Text recognition for iOS uses Firebase MLKit which requires setting up Firebase project for your app.
```

**Comment by n1ru4l** [comment]
We definitely need a sample project to ensure everything is working correct.

**Comment by dov11** [comment]
@n1ru4l, I have added an example using mlkit, currently with `google-services.json`/`GoogleService-Info.plist` in `.gitignore`. So if a user wants to test it they can create their own Firebase project and add the files to the example (described in example README). I'm not sure how safe it is to add the files to source control, so maybe for the CI a dummy file should be created with values overwritten at build time with environment variables (I think I would need to leave that to maintainers)?

**Comment by sibelius** [comment]
can you rebase, I'm going to run example on both android and ios so we can merge this soon

**Comment by sibelius** [comment]
android is working fine both camera features, text detection and face detection

on ios I'm receiving this warning

> FaceDetector not integrated, stub used!

maybe because the `mlkit` example is not using old ios face detection code

I think we can merge this after rebase/conflicts fixed and do a major bump

tks for the great work on this 💯

**Comment by dov11** [comment]
@sibelius , thanks for testing and reviewing the PR, indeed currently `mlkit` example does not have a FaceDetector integrated - podspec has 'FaceDetector' podspec commented out. I have started migrating facedetector to mlkit for ios as well, and will have podspec reactivated once the migration is complete.

I have rebased the PR on the recent master and have amended mlkit and basic examples to include barcode recognition - i think this PR fixes the issue that this [PR](https://github.com/react-native-community/react-native-camera/pull/2095#issuecomment-461351242) was still having with bounds detection and you can test it now.
I also reverted inclusion of `react-navigation` and `react-native-gesture-handler` in basic example as we can can demonstrate all capabilities without the need of managing two more dependencies.

**Comment by sibelius** [comment]
Great. I’ll test again 

Should we wait until you migrate face detection on iOS?

I think we can have another example with react navigation, as having 2 camera components mounted can cause trouble

**Comment by dov11** [comment]
I don't think this PR needs to wait just for ios face detection - I can bring it in separately, the rest seems to be working fine without it. On the other hand, quite a few files have been touched in the PR so rebasing it after new features are introduced becomes a bit of a chore.

After the amendment, examples do not require 2 camera components - all features are demonstrated with one.

**Comment by dov11** [comment]
@sibelius, I just noticed a problem introduced by #2007 - duplicate method bridgeDidBackground:
[1](https://github.com/reime005/react-native-camera/blob/2d19d5f51ed2bd8b7c77d9e629b2913c6073e847/ios/RN/RNCamera.m#L808) and [2](https://github.com/reime005/react-native-camera/blob/2d19d5f51ed2bd8b7c77d9e629b2913c6073e847/ios/RN/RNCamera.m#L824)
I pushed a fix into the branch of this PR, but you should know that the current master might be broken

File: README.md [hunk 2]
```diff
@@ -323,6 +366,38 @@ allprojects {
...
+    7.2. Configure your app to automatically download the ML model to the device after your app is installed from the Play Store. If you do not enable install-time model downloads, the model will be downloaded the first time you run the on-device detector. Requests you make before the download has completed will produce no results. 
+
+    ```xml
+    <application ...>
+    ...
+      <meta-data
+          android:name="com.google.firebase.ml.vision.DEPENDENCIES"
+          android:value="ocr" />
+      <!-- To use multiple models, list all needed models: android:value="ocr, face, barcode" -->
+    </application>
```
Comment (mieszko4) [inline README.md]: this applies only to `face`/`ocr`

File: README.md [hunk 3]
```diff
@@ -323,6 +366,38 @@ allprojects {
 }
 ```
 
+7. Additional steps for using MLKit for text/face/barcode recognition
```
Comment (mieszko4) [inline README.md]: I've noticed that qr code recognition works for me on android without following instructions for integration of firebase

**Pull request merged**